src/og/Events.ts

```diff
@@ -1,53 +1,66 @@
 "use strict";
 
-import { binaryInsert, stamp } from "./utils/shared.js";
+import {binaryInsert, stamp} from "./utils/shared";
+
+type EventCallbacks = Array<() => void>;
+
+export type EventsMap<T extends string[]> = {
+    [K in T[number]]?: { active: boolean; handlers: EventCallbacks }
+}
+
+export type EventsHandler<T extends string[]> = Events<T> & EventsMap<T>;
+
+export function createEvents<T extends string[]>(methodNames: T, sender?: any) {
+    return new Events(methodNames, sender) as EventsHandler<T>;
+}
 
 /**
  * Base events class to handle custom events.
  * @class
+ * @param {Array.<string>} [eventNames] - Event names that could be dispatched.
+ * @param {*} [sender]
  */
-class Events {
+export class Events<T extends string[]> {
+
     /**
-     *
-     * @param {Array.<string>} [eventNames] - Event names that could be dispatched.
-     * @param {*} [sender]
+     * Registered event names.
+     * @protected
+     * @type {Array.<string>}
+     */
+    protected _eventNames: T;
+
+
+    protected _sender: any;
```

Comment:
I agree with sender any - it's could be any object from system
But disagree about `callback: any`! It's function, whith can get array of any parameters and return void or boolean
Disagree about _stampCache - it's Record<string, callback> 
And `as any` looks strange

propose decline any by eslint rools - it's a bad practice. Where we defenetly need any can off this rool by comment

Reply:
Also I think we need to declarate somehow created events in this class. this.event[event_name] shouldn't be `any`

Reply:
You could help! :)